docs/docs/user-permissions/access-control.md

```diff
@@ -0,0 +1,147 @@
+
+:::info
+By default COMET DXP tries to load the affected entity by id with the value of the submitted id-argument. However, the name of the argument can be altered by using the `idArg` setting.
+:::
+
+```ts
+@Query(Product)
+@AffectedEntity(Product)
+async product(@Args("id", { type: () => ID }) id: string): Promise<Product> {
+    //...
+}
+```
+
+```ts
+@Query([Product])
+@AffectedEntity(Dealer, { idArg: "dealerId" })
+async products(@Args("dealerId", { type: () => ID }) dealerId: string): Promise<Product[]> {
+    // Note: you can trust "dealerId" being in a valid scope, but you need to make sure that your business code restricts this query to the given dealer
+}
+```
+
+**@ScopedEntity**
+
+Retrieving the affected entity alone is not sufficient, COMET DXP also needs to know the scope of the entity. The simplest case is when the entity has a field named `scope`. If this is true, this decorator is not necessary.
+
+If the scope is stored in a different field or the entity has a relation to another entity that stores the scope, additional information is required. This is where this decorator comes into play.
+
+Use this decorator at the **entity level** to return the scope of an entity.
+
+```ts
+@ScopedEntity(async (product: Product) => {
+    return {
+        dealer: product.dealer.id,
+    };
+})
+@Entity()
+export class Product extends BaseEntity<Product, "id"> {}
+```
+
+:::info
+You might have to load multiple relations for nested data.
+:::
+
+## Disable permission/scope checks
+
+**skipScopeCheck**
+
+The scope check can be disabled by adding `{skipScopeCheck: true}` as the second argument of the `@RequiredPermission` decorator.
+
+:::caution
+Use this option only when you are sure that checking the scope is not necessary (e.g. the current entity does not have a scope). Do not add it just because it seems cumbersome at the moment to add the correct `AffectedEntity`/`ScopedEntity` decorators.
+:::
+
+:::note
+Try to avoid using the `getCurrentUser` decorator. Instead, you should explicitly send all the data needed in an operation. In the following example, this requires adding `userId` as a scope part as well as passing the data throughout the client. Nevertheless, this leads to a cleaner API design.
+
+```diff
+- @RequiredPermission("products", {skipScopeCheck: true})
++ @RequiredPermission("products")
++ @AffectedEntity(User)
+- async myProducts(@GetCurrentUser() currentUser: CurrentUser): Promise<Product[]> {
++ async productsForUser(@Args("userId", { type: () => ID }) userId: string): Promise<Product[]> {
+      //...
+  }
+```
+
+:::
+
+**@PublicApi**
+
+`@PublicApi()` can be used to expose a single handler (query, mutation or route) or a whole class (resolver or controller) publicly.
+
+:::caution
+
+Using the decorator at class level causes later added handlers to be automatically public. Prefer using the decorator for single handlers only.
```

Comment:
```suggestion
Using the decorator at class level causes later added operations to be automatically public. Prefer using the decorator for single operations only.
```

Since we always use the term "operation" now